flocker/provision/_ssh/_conch.py

```diff
@@ -87,6 +90,59 @@ def lineReceived(self, line):
         ).write()
 
 
+class CommandChannelWithTTY(_CommandChannel):
+    """
+    CentOS/RHEL wants us to have a pty in order to run commands with sudo.
+    Create a pty that won't be used when creating the channel.
+    """
+    def channelOpen(self, ignored):
+        """
+        Create a pty by sending a pty-req to the server
+        """
+        term = 'xterm'
+        winSize = (25, 80, 0, 0)
+        ptyReqData = session.packRequest_pty_req(term, winSize, '')
+        self.conn.sendRequest(self, 'pty-req', ptyReqData)
+        command = self.conn.sendRequest(
+            self, 'exec', NS(self._command), wantReply=True)
+        command.addCallbacks(self._execSuccess, self._execFailure)
+
+
+class SSHCommandClientEndpointWithTTY(SSHCommandClientEndpoint):
+    """
+    Subclass that spawns a TTY when connecting.
+    """
+    def _executeCommand(self, connection, protocolFactory):
+        """
+        Given a secured SSH connection, try to execute a command in a new
+        channel created on it and associate the result with a protocol from the
+        given factory.
+
+        @param connection: See L{SSHCommandClientEndpoint.existingConnection}'s
+            C{connection} parameter.
```

Comment:
This is actually copy-pasted directly from SSHCommandClientEndpoint's implementation.  I could delete the docstring.  Do you think that would be better?